src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -342,6 +343,11 @@ private TypeSpec CreateArraySpec(TypeParseInfo typeParseInfo)
                     return CreateUnsupportedTypeSpec(typeParseInfo, NotSupportedReason.MultiDimArraysNotSupported);
                 }
 
+                if (IsUnsupportedType(typeSymbol.ElementType))
```

Comment:
Am I right to assume that array types are special cases because they follow a separate code path than the `CreateEnumerableSpec` method?

Reply:
We talked offline. Yes this is special casing the array case.